lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -663,8 +663,24 @@ sub done {
     $validation->optional('result');
     $validation->optional('reason');
     $validation->optional('newbuild')->num(1);
+    $validation->optional('worker_id')->num(0, undef);
     return $self->reply->validation_error({format => 'json'}) if $validation->has_error;
 
+    # check whether the specified worker matches the actually assigned one; refuse the update if not
+    if (my $specified_worker_id = $validation->param('worker_id')) {
+        my $assigned_worker_id = $job->assigned_worker_id;
+        my $msg                = (
+            defined $assigned_worker_id
+            ? (
+                $assigned_worker_id != $specified_worker_id
+                ? "Refusing to set result because job is now assigned to worker $assigned_worker_id)."
+                : undef
```

Comment:
Trying to re-phrase my previous question. The code is not touching the job, right? Which means *now* is misleading if the code is merely looking at the current state.

Reply:
Perhaps you mean to say "because the job is *currently* assigned to worker"?

Reply:
And finally, if that's true, something like "Jobs cannot be moved to different workers at runtime" might provide an explanation for why this is considered an error.

Reply:
I changed it from "now" to "currently" but I doubt a lot of people know/care about the difference.

Reply:
I've just seen your last comment but I don't think adding that explanation is helpful. Nobody it trying to move jobs between workers. Note that this error message will only end up on the worker log (as API error reported by the web UI) and won't be user-visible at all. So I wouldn't put too much effort into it.